### PR TITLE
Repository class

### PR DESCRIPTION
**Pull request by MaximilianV**
This PR should fix several issues:
 - To many unnecessary requests were made
 - Lanes didn't update properly #14 
 - Magically there were several masters after pushing a new branch 🦄

### CODE REVIEW

File: src/Overview/ProjectLane/BranchCard/BranchCard.js [hunk 1]
```diff
@@ -4,73 +4,85 @@ import ProgressLane from './CardLanes/ProgressLane/ProgressLane.js';
...
-    super(props);
-    this.state = {branchName: this.props.branch.getName()};
-    this.branch = this.props.branch;
-  }
-  componentDidMount() {
-    if(this.branch.isIssueExtractable()) {
-      this.setState({issueNumber: this.branch.getIssueNumber(), issueDescription: this.branch.getIssueName()});
+    constructor(props) {
+        super(props);
+        this.state = {branchName: this.props.branch.getName(),
```
Comment (friedow) [inline src/Overview/ProjectLane/BranchCard/BranchCard.js]: Please break this line and fix the indentation.

File: src/Overview/ProjectLane/BranchCard/BranchCard.js [hunk 2]
```diff
@@ -4,73 +4,85 @@ import ProgressLane from './CardLanes/ProgressLane/ProgressLane.js';
...
-  getTitleForCard() {
-    if (this.state.issueNumber && this.state.issueDescription) {
-      return (
-        <span>
+    extractInformation(props) {
+        if (this.state.branchName !== props.branch.getName()) {
+            this.setState({branchName: props.branch.getName()});
+        }
+        if(props.branch.isIssueExtractable()) {
+            this.setState({issueNumber: props.branch.getIssueNumber(), issueDescription: props.branch.getIssueName()});
```
Comment (friedow) [inline src/Overview/ProjectLane/BranchCard/BranchCard.js]: Please break this line after { and ,

File: src/Overview/ProjectLane/BranchCard/BranchCard.js [hunk 3]
```diff
@@ -4,73 +4,85 @@ import ProgressLane from './CardLanes/ProgressLane/ProgressLane.js';
...
+        }
     }
-  }
 
-  getColorBasedOnLatestBuild() {
-    if(!this.branch.didTravisRun()) {
-      return "grey darken-2";
+    getColorBasedOnLatestBuild() {
+        if(!this.state.branch.didTravisRun()) {
+            return "grey darken-2";
```
Comment (friedow) [inline src/Overview/ProjectLane/BranchCard/BranchCard.js:51]: We should consider moving color definitions to a config file

File: src/Overview/ProjectLane/BranchCard/BranchCard.js [hunk 4]
```diff
@@ -4,73 +4,85 @@ import ProgressLane from './CardLanes/ProgressLane/ProgressLane.js';
...
-              <ProgressLane value={this.branch.getCoverage()} valueText={this.branch.getNiceCoverage()} changeText={this.branch.getNiceCoverageChange()} showLabels={false} />
-              <UserLane username={this.branch.getLatestCommitter()} lastCommitTime={this.branch.getLastCommitTimeAsString()} avatarUrl={this.branch.getLatestCommitterAvatarUrl()}/>
+    render() {
+        return (
+            <div className="branch-card col s2" key={this.state.branchName} >
+                <div className={`card small horizontal hoverable ${this.getColorBasedOnLatestBuild()}`}>
+                    <div className="card-content white-text">
+                        <span className="card-title">{this.getTitleForCard()}</span>
+                        <div className="status-lanes">
+                            <TextLane value={`</> ${this.state.branch.getNiceSonarqubeString()}`} color={this.getColorBasedOnQualityGate()} />
```
Comment (friedow) [inline src/Overview/ProjectLane/BranchCard/BranchCard.js]: Please break this and the following 2 lines. 
Corresponding code convention: If a component initialises with 2 or more parameters, insert a line break before each parameter.

**Review by friedow — CHANGES_REQUESTED** [review]

File: src/Overview/ProjectLane/BranchCard/BranchCard.js (hunk 3 above)
Comment (MaximilianV) [reply src/Overview/ProjectLane/BranchCard/BranchCard.js:51]: You are right, but I didn't want to do this in this refactoring branch :) But we should definitely consider some constants here.

**Review by friedow — APPROVED** [review]

**Pull request merged**